packages/input-segments/src/InputSegments.ts

```diff
@@ -0,0 +1,1287 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import {
+    CalendarDateTime,
+    DateFormatter,
+    endOfMonth,
+    getLocalTimeZone,
+    getMinimumDayInMonth,
+    getMinimumMonthInYear,
+    now,
+    toCalendarDateTime,
+} from '@internationalized/date';
+import { NumberParser } from '@internationalized/number';
+import {
+    CSSResultArray,
+    html,
+    PropertyValueMap,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import {
+    property,
+    query,
+    state,
+} from '@spectrum-web-components/base/src/decorators.js';
+import {
+    ClassInfo,
+    classMap,
+    ifDefined,
+    StyleInfo,
+    styleMap,
+    when,
+} from '@spectrum-web-components/base/src/directives.js';
+import { LanguageResolutionController } from '@spectrum-web-components/reactive-controllers/src/LanguageResolution.js';
+import { TextfieldBase } from '@spectrum-web-components/textfield';
+
+import {
+    AM,
+    dateSegmentTypes,
+    maxHourAM,
+    maxHourPM,
+    minHourAM,
+    minHourPM,
+    PM,
+    Segment,
+    SegmentDetails,
+    SegmentValueAndLimits,
+    TimeGranularity,
+    timeSegmentTypes,
+} from './types.js';
+
+import styles from './input-segments.css.js';
+
+/**
+ * An utility to check if the given value is a number (not `undefined`)
+ *
+ * @param value - Number to check
+ */
+const isNumber = (value: number | undefined): value is number => {
+    return typeof value === 'number';
+};
+
+/**
+ * @event change - Announces when a new date/time is defined by emitting a `Date` object
+ *
+ * @slot help-text - Default or non-negative help text to associate to your form element
+ * @slot negative-help-text - Negative help text to associate to your form element when `invalid`
+ */
+export class InputSegments extends TextfieldBase {
+    public static override get styles(): CSSResultArray {
+        return [...super.styles, styles];
+    }
+
+    /**
+     * Indicates when date segments should be included in the field
+     */
+    @state()
+    protected includeDate = false;
+
+    /**
+     * Indicates when time segments should be included in the field
+     */
+    @state()
+    protected includeTime = false;
+
+    /**
+     * Indicates which segments that are part of time should be used
+     */
+    @property()
+    timeGranularity: TimeGranularity = 'minute';
```

Comment:
It doesn't exist because we don't need one. Using a granularity as your example would work more or less "from left to right", but if I specify the granularity as 'hour' it doesn't mean that I want to include the date part as well, as is the case with the time field. If I pass a 'date' granularity to the time field, nothing would be rendered, so I thought I'd leave this property specifically for when we use time, and 'hour' would always be the minimum that would be displayed. What do you think?